eng/common/scripts/Package-Properties.ps1

```diff
@@ -164,21 +148,45 @@ function Get-CSVMetadata ([string]$MetadataUri=$MetadataUri)
     return $metadataResponse
 }
 
-function Operate-OnPackages ($activePkgList, $ServiceDirectory, [Array]$pkgPropsResult)
+function Get-PkgPropsForEntireService ($serviceDirectoryPath)
 {
-    foreach ($pkg in $activePkgList)
+    $projectProps = @() # Properties from very project inthe service
+    $packageProps = @() # Properties for artifacts specified in ci.yml
+
+    foreach ($directory in (Get-ChildItem $serviceDirectoryPath -Directory))
     {
-        LogDebug "Operating on $($pkg["name"])"
-        $pkgProps = Get-PkgProperties -PackageName $pkg["name"] -ServiceDirectory $ServiceDirectory
-        if ($null -ne  $pkgProps)
+        $pkgDirectoryPath = Join-Path $serviceDirectoryPath $directory.Name
+
+        if ($GetPackageInfoFromRepoFn -and (Test-Path "Function:$GetPackageInfoFromRepoFn"))
         {
-            $pkgPropsResult += $pkgProps
+            $pkgProps = &$GetPackageInfoFromRepoFn $pkgDirectoryPath (Split-Path -Path $serviceDirectoryPath -Leaf)
```

Comment:
Lets capture the serviceDirectory once outside of the look instead of doing it everytime through the loop. We should probably do that with the function as well.